bat/tests/06-test-format-bump/run.bats

```diff
@@ -1,14 +1,48 @@
 #!/usr/bin/env bats
 
 # shared test functions
-#load ../../lib/mixerlib
+load ../../lib/mixerlib
 
-#setup() {
-#  global_setup
-#}
+setup() {
+  global_setup
+}
 
-# TODO: fill out once auto format bump implemented
-#@test "Create initial mix 10" {
-#}
+# TODO: This test should run 'mixer build upstream-format' instead
+# once the tooling required to run custom test docker images for
+# is ready.
+@test "Perform upstream format bump on last format boundary" {
+  upstreamver=$(get-last-format-boundary)
+  mixer-init-stripped-down $upstreamver 10
+
+  #Create initial version in the old format
+  mixer-build-all > $LOGDIR/build_all.log
+
+  #Update to new upstream in the new format
+  mixer-upstream-update $(($upstreamver + 10)) > $LOGDIR/upstream_update.log
+
+  #Build +20
+  mixer-build-format-bump-new > $LOGDIR/build_formatbump_new.log
+
+  #Reset upstream for +10 build
+  echo $upstreamver > upstreamversion
+
+  #Build +10
+  mixer-build-format-bump-old > $LOGDIR/build_formatbump_old.log
+
+  #restore upstrean
+  mv upstreamversion.bump upstreamversion
+
+  #check if LAST_VER is set to +20
+  test $(cat update/image/LAST_VER) -eq 30
+
+  #check if +20 previous version is set to +0
+  test $(awk -v FS="previous:" 'NF>1{gsub(/\t/, "", $2); print $2}' update/www/30/Manifest.MoM) -eq 10
+
+  #check if +10 previous version is set to +0
+  test $(awk -v FS="previous:" 'NF>1{gsub(/\t/, "", $2); print $2}' update/www/20/Manifest.MoM) -eq 10
```

Comment:
While this works, I would code this as

```awk '$1 == "previous:" { exit $2 == 10}'```

which I find much more obvious. Setting the FS to previous: introduces a corner case if the string previous: occurs anywhere on the line, perhaps as part of a filename.

Reply:
Indeed it looks better. I changed it to 'exit $2 != 10' since I want it to return 0 if they are the same.